client/transport.c

```diff
@@ -353,12 +353,15 @@ static int do_request(int type, xtime_t start_time, xtime_t end_time, void *data
 	expire_time.tv_sec = actual_time + REQUEST_WAIT;
 
 	while(!reply_rcvd){
-		if(pthread_cond_timedwait(&g_process_cond, &g_process_mutex, &expire_time)) {
+		errno = 0;
+		if((ret = pthread_cond_timedwait(&g_process_cond, &g_process_mutex, &expire_time))) {
 			last_reply_id = reply_id_private;
 			reply_data = NULL;
 			reply_callback = NULL;
-			if(errno != EAGAIN || errno != ETIMEDOUT) {
-				xdag_err("pthread_cond_timedwait failed [function: do_request]");
+			if(errno && (errno != EAGAIN || errno != ETIMEDOUT)) {
```

Comment:
Do need || here? (errno != EAGAIN || errno != ETIMEDOUT) is always true.

Reply:
how can be always true? errno may be EAGAIN for example, or EINVAL
on some system errno is set and in others err code is returned from the function so i check both

Reply:
I removed it anyway since by standard error number whould be returned from the function